### ISSUE TITLE
Validate MIDI note range

### ISSUE THREAD

**Issue by daveyarwood** [alda-lang/alda#321]
Currently, attempting to play a note that is out of the MIDI note range (0-127) results in a confusing error message:

```
$ alda play -c 'piano: o10 c'
[27713] Parsing/evaluating...
[27713] ERROR data1 out of range: 132
```

We should catch this and throw our own error with a clearer message.

**Comment by daveyarwood** [comment]
I just tried this with Alda 2, and interestingly, playback seems to just silently fail once it gets up to the too-high note. No error message is printed either in the client or in the player logs.

So the disposition is still the same: we should add validation that the MIDI note isn't < 0 or > 127, and I think we should do that validation at score evaluation time instead of playback-time, so that we can display a helpful message earlier.

I think the place to add the validation now is [right here](https://github.com/alda-lang/alda/blob/48ea575aad588c1288116b8289a865af9231583b/client/model/note.go#L120-L122) after we calculate the MIDI note based on the octave, note letter and accidentals.

**Comment by KartikYZ** [comment]
I have added the validation at the specified place and submitted a PR in response to this issue. Also ran the addition against the test suite.

**Comment by daveyarwood** [comment]
Fixed in https://github.com/alda-lang/alda/pull/461. Will keep this issue open until I have a chance to release it.

**Comment by daveyarwood** [comment]
Fixed in Alda 2.2.4.
